addons/account/models/res_partner_bank.py

```diff
@@ -70,7 +70,7 @@ def _compute_duplicate_bank_partner_ids(self):
         id2duplicates = dict(self.env.execute_query(SQL(
             """
                 SELECT this.id,
-                       ARRAY_AGG(other.partner_id)
+                       ARRAY_AGG(other.partner_id) FILTER (WHERE other.partner_id IS NOT NULL)
                   FROM res_partner_bank this
              LEFT JOIN res_partner_bank other ON this.acc_number = other.acc_number
```

Comment:
Could we just do this instead of adding the `FILTER`?
```suggestion
                  JOIN res_partner_bank other ON this.acc_number = other.acc_number
```
I think there is no need to join missing "`other`" partner banks in the first place.
With your fix we would have an entry `{1: None}` in the dictionary in case of no duplicates for partner bank with id `1`. With `JOIN` we would just omit such entries in the dictionary. But since we access the dictionary with `get` anyway it should be the same.

Reply:
I think we don't need to add the `FILTER` then is what I meant.
Or is it necessary for some reason?